src/esssans/beam_center_finder.py

```diff
@@ -86,11 +86,18 @@ def beam_center_from_center_of_mass(
         The beam center position as a vector.
     """
 
-    # TODO: the flattening is required for the selection with a boolean mask to work.
-    # We could avoid the copy made here by working with the underlying numpy arrays.
-    data = data.flatten(dims=data.coords['position'].dims, to=uuid.uuid4().hex)
-    pos = data.coords['position']
-    summed = data.bins.sum()
+    dims_to_sum = set(data.dims) - set(data.coords['position'].dims)
+    if dims_to_sum:
+        summed = data.sum(dims_to_sum)
+    else:
+        summed = data.bins.sum()
+    if summed.ndim > 1:
+        # TODO: the flattening is required for the selection with a boolean mask to
+        # work. We could avoid the copy made here by working with the underlying numpy
+        # arrays.
```

Comment:
What copy? Numpy boolean indexing cannot return without making a copy either?

Reply:
ah yes, I think that's correct. I'll remove the comment.